Core/include/Acts/Utilities/TransformRange.hpp

```diff
@@ -0,0 +1,133 @@
+  constexpr static value_type apply(const input_t& value) { return *value; }
+};
+
+template <template <typename> typename Callable, typename iterator_t,
+          bool force_const>
+struct TransformRangeIterator {
+ private:
+  using internal_value_type = typename iterator_t::value_type;
+
+ public:
+  using value_type = std::conditional_t<
+      force_const,
+      decltype(std::as_const(
+          std::declval<typename Callable<internal_value_type>::value_type>())),
+      typename Callable<internal_value_type>::value_type>;
+
+  explicit TransformRangeIterator(iterator_t iterator) : m_iterator(iterator) {}
+
+  value_type operator*() {
+    return Callable<typename iterator_t::value_type>::apply(*m_iterator);
+  }
+
+  value_type operator*() const {
+    return Callable<typename iterator_t::value_type>::apply(*m_iterator);
+  }
+
+  TransformRangeIterator& operator++() {
+    ++m_iterator;
+    return *this;
+  }
+
+  bool operator==(const TransformRangeIterator& other) const {
+    return m_iterator == other.m_iterator;
+  }
+
+ private:
+  iterator_t m_iterator;
+};
+
+template <template <typename> typename Callable, typename container_t>
+struct TransformRange {
+ private:
+  using internal_value_type = typename container_t::value_type;
+
+ public:
+  using value_type = std::conditional_t<
+      std::is_const_v<container_t>,
+      decltype(std::as_const(
+          std::declval<typename Callable<internal_value_type>::value_type>())),
+      typename Callable<internal_value_type>::value_type>;
+
+  using iterator = TransformRangeIterator<
+      Callable,
+      std::conditional_t<std::is_const_v<container_t>,
+                         typename container_t::const_iterator,
+                         typename container_t::iterator>,
+      std::is_const_v<container_t>>;
+  using const_iterator =
+      TransformRangeIterator<Callable, typename container_t::const_iterator,
+                             true>;
+
+  explicit TransformRange(container_t& container) : m_container(&container) {}
+
+  value_type operator[](std::size_t i) {
+    return Callable<internal_value_type>::apply((*m_container)[i]);
+  }
+
+  decltype(auto) operator[](std::size_t i) const {
+    return std::as_const(
+        Callable<internal_value_type>::apply((*m_container)[i]));
+  }
+
+  value_type at(std::size_t i) {
+    return Callable<internal_value_type>::apply(m_container->at(i));
+  }
+
+  decltype(auto) at(std::size_t i) const {
+    return std::as_const(
+        Callable<internal_value_type>::apply(m_container->at(i)));
+  }
+
+  iterator begin() { return iterator{m_container->begin()}; }
+
+  iterator end() { return iterator{m_container->end()}; }
+
+  const_iterator begin() const { return const_iterator{m_container->begin()}; }
+
+  const_iterator end() const { return const_iterator{m_container->end()}; }
+
+  std::size_t size() const { return m_container->size(); }
+  bool empty() const { return m_container->empty(); }
+
```

Comment:
Do you want to add ```cbegin()```  for this class? The other stuff  ```back(), front()``` etc is probably not so important

Reply:
I guess I can just implement `cbegin` / `cend` like

```cpp
const_iterator cbegin() const { return begin(); }
```